### PR TITLE
[1152] Add support for form in the view converter

### PR DESCRIPTION
**Pull request by florianbarbin**
### Type of this PR 

- [ ] Bug fix
- [x] New feature or improvement
- [ ] Documentation
- [ ] Cleanup
- [ ] Test
- [ ] Build/Releng

### Issue(s)
#1152 
...

### What does this PR do?

Refactor the ViewConverter to handle each kind of RepresentationDescription.
Add support for form in the view converter. We only handle a simple form with a read only textfield in this first version.
...

### Screenshot/screencast of this PR

...
 

### Potential side effects

...

### How to test this PR?

- [ ] Frontend Unit tests
- [x] Backend Unit tests
- [ ] Cypress : please specify
- [ ] Manual Test : please specify

### Checklist

- [ ] I have read CONTRIBUTING carefully.
- [x] I have signed the [Contributor License Agreement](http://www.eclipse.org/legal/CLA.php).
- [x] All my commits are signed-off (`-s`) with my mail address of my Eclipse Account.
- [ ] I have covered my changes by unit tests or integration tests or manual tests.
- [ ] All tests pass.
- [ ] I have updated the documentation accordingly
- [ ] New React components are availables in storybook

### CODE REVIEW

**Review by sbegaudeau — CHANGES_REQUESTED** [review]
- Fix the title of your PR
- Ensure that your PR is in the proper project and milestone
- Ensure that your PR has the proper labels (look at other PRs to see what is being done)

File: backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/ViewConverter.java [hunk 1]
```diff
@@ -54,10 +52,11 @@ public List<IRepresentationDescription> convert(View view, List<EPackage> visibl
         try {
             // @formatter:off
             result = view.getDescriptions().stream()
-                         .filter(org.eclipse.sirius.components.view.DiagramDescription.class::isInstance)
-                         .map(org.eclipse.sirius.components.view.DiagramDescription.class::cast)
-                         .map(viewDiagramDescription -> this.convertDiagramDescription(viewDiagramDescription, interpreter))
-                         .collect(Collectors.toList());
+                    .map(representationDescription -> this.convert(representationDescription, interpreter))
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
```
Comment (sbegaudeau) [inline backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/ViewConverter.java:57]: flatMap(Optional::stream)

File: backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/ViewConverter.java [hunk 2]
```diff
@@ -66,6 +65,16 @@ public List<IRepresentationDescription> convert(View view, List<EPackage> visibl
         return result;
     }
 
+    private Optional<IRepresentationDescription> convert(RepresentationDescription representationDescription, AQLInterpreter aqlInterpreter) {
+        // @formatter:off
+        return this.representationDescriptionConverters.stream()
+                .filter(converter -> converter.canConvert(representationDescription))
+                .map(converter -> converter.convert(representationDescription, aqlInterpreter))
+                .findFirst()
+                .orElse(Optional.empty());
```
Comment (sbegaudeau) [inline backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/ViewConverter.java:74]: remove the `orElse(Optional.empty())`

File: backend/sirius-components-emf/src/test/java/org/eclipse/sirius/components/emf/view/DynamicFormsTests.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    void buildFixture() {
+        this.fixture = EcoreFactory.eINSTANCE.createEPackage();
+        this.fixture.setName("fixture"); //$NON-NLS-1$
+        EClass klass1 = EcoreFactory.eINSTANCE.createEClass();
+        klass1.setName("Class1"); //$NON-NLS-1$
+        this.fixture.getEClassifiers().add(klass1);
+        EClass klass2 = EcoreFactory.eINSTANCE.createEClass();
+        klass2.setName("Class2"); //$NON-NLS-1$
+        this.fixture.getEClassifiers().add(klass2);
+    }
```
Comment (sbegaudeau) [inline backend/sirius-components-emf/src/test/java/org/eclipse/sirius/components/emf/view/DynamicFormsTests.java:63]: Don't start making stateful test, it only gets worse from there. Copy a static ecore file in the test resources for example if you want a meaningful test.

File: backend/sirius-components-emf/src/test/java/org/eclipse/sirius/components/emf/view/DynamicFormsTests.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for dynamically defined forms.
+ *
+ * @author fbarbin
+ */
+public class DynamicFormsTests {
+
+    private static final String NAME_EXPRESSION = "aql:self.name"; //$NON-NLS-1$
```
Comment (sbegaudeau) [inline backend/sirius-components-emf/src/test/java/org/eclipse/sirius/components/emf/view/DynamicFormsTests.java:49]: In unit test, you can use the `@SuppressWarnings("checkstyle:MultipleStringLiterals")` and not create constants for things like that. It does not make tests easier to read.

**Review by sbegaudeau — CHANGES_REQUESTED** [review]

File: backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/ViewConverter.java (hunk 2 above)
Comment (florianbarbin) [reply backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/ViewConverter.java:74]: converter.convert returns an Optional. I will replace by 
                .flatMap(Optional::stream)
                .findFirst();

File: backend/sirius-components-emf/src/test/java/org/eclipse/sirius/components/emf/view/DynamicFormsTests.java (hunk 3 above)
Comment (florianbarbin) [reply backend/sirius-components-emf/src/test/java/org/eclipse/sirius/components/emf/view/DynamicFormsTests.java:63]: Fixed

File: backend/sirius-components-emf/src/test/java/org/eclipse/sirius/components/emf/view/DynamicFormsTests.java (hunk 4 above)
Comment (florianbarbin) [reply backend/sirius-components-emf/src/test/java/org/eclipse/sirius/components/emf/view/DynamicFormsTests.java:49]: Fixed.

File: backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/ViewConverter.java (hunk 1 above)
Comment (florianbarbin) [reply backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/ViewConverter.java]: done

**Review by sbegaudeau — CHANGES_REQUESTED** [review]

**Comment by sbegaudeau** [comment]
By the way, is there any entry in the Kiwi TCMS on how to test that feature?

**Comment by florianbarbin** [comment]
> By the way, is there any entry in the Kiwi TCMS on how to test that feature?



> By the way, is there any entry in the Kiwi TCMS on how to test that feature?

Not yet.

**Review by sbegaudeau — APPROVED** [review]
I'll merge that after the build

**Pull request merged**